File: training-data/go/job_sequencing.go
package main

import (
	"fmt"
	"sort"
)

type job struct {
	id               string
	deadline, profit int
}

func schedule(jobs []job) ([]string, int) {
	s := append([]job(nil), jobs...)
	sort.Slice(s, func(i, j int) bool { return s[i].profit > s[j].profit })
	maxD := 0
	for _, j := range s {
		maxD = max(maxD, j.deadline)
	}
	slots := make([]string, maxD+1)
	total := 0
	for _, j := range s {
		for t := j.deadline; t > 0; t-- {
			if slots[t] == "" {
				slots[t] = j.id
				total += j.profit
				break
			}
		}
	}
	var order []string
	for _, id := range slots[1:] {
		if id != "" {
			order = append(order, id)
		}
	}
	return order, total
}

func main() {
	fmt.Println(schedule([]job{{"a", 2, 100}, {"b", 1, 19}, {"c", 2, 27}, {"d", 1, 25}, {"e", 3, 15}}))
}

File: training-data/typescript/job_sequencing.ts
export interface Job { id: string; deadline: number; profit: number }

class DSU {
  private parent: number[];
  constructor(n: number) {
    this.parent = Array.from({ length: n }, (_, i) => i);
  }
  find(x: number): number {
    while (this.parent[x] !== x) x = this.parent[x] = this.parent[this.parent[x]];
    return x;
  }
  link(x: number, y: number): void {
    this.parent[x] = y;
  }
}

export function sequenceJobs(jobs: readonly Job[]): { order: string[]; profit: number } {
  const maxD = Math.max(...jobs.map((j) => j.deadline));
  const dsu = new DSU(maxD + 1);
  const slots: (Job | undefined)[] = new Array(maxD + 1);
  for (const job of [...jobs].sort((a, b) => b.profit - a.profit)) {
    const free = dsu.find(Math.min(job.deadline, maxD));
    if (free > 0) {
      slots[free] = job;
      dsu.link(free, free - 1);
    }
  }
  const chosen = slots.filter((j): j is Job => j !== undefined);
  return { order: chosen.map((j) => j.id), profit: chosen.reduce((s, j) => s + j.profit, 0) };
}

console.log(sequenceJobs([
  { id: "a", deadline: 2, profit: 100 },
  { id: "b", deadline: 1, profit: 19 },
  { id: "c", deadline: 2, profit: 27 },
  { id: "d", deadline: 1, profit: 25 },
  { id: "e", deadline: 3, profit: 15 },
]));
